CenterForOpenScience/osf.io
# Anonymized view only link public view changes

## Purpose
Previously: Public projects in anonymized view only links showed all information
Now: Only projects selected when creating a view only link will be visible.

## Changes
Adds to the node.can_view logic to see if a project is being viewed through an anonymous private link. If it is, only projects selected when the private link was created will be viewable

Adds private_link to auth which gives access to the current private links details from auth without needing to explicitly look it up each time 

## Side Effects
1. Because this changes what can be viewed there are many places that side effects will occur in both hiding content that should not be seen, and showing content that should not be seen. I believe it is working properly, but there are lots of combos that may need to be tested to ensure they all work properly. 

2. Anonymous view only links anonymize indiscriminately. Navigating to another project with the view only link will anonymize that project, the dashboard, everything. It shouldn't be too hard to fix this, but wanted it noted that this occurs.

## QA Considerations
For each auth situation, the files, components, and recent activity should show only the proper information, and be properly anonymized. 

Combinations of these variables should be tested, some may be excluded where overlap occurs:
1. Logged in (any read/write access), Not logged in
2. Anonymized private link, Not anonymized private link, Non existent private link, Private link from another project 
3. Public Project, Private Project
In the project:
4. Public Node (checked), Public Node (unchecked), Private Node (checked), Private Node (unchecked)

When there is an anonymous private link, this overrides all other situations. 
1. Nodes included in the anonymous private link should be visible
2. Nodes not included in the anonymous private link should not be visible

Auth should work as expected in all other situations.
